lib/elixir/lib/macro/env.ex

```diff
@@ -185,6 +185,39 @@ defmodule Macro.Env do
   def fetch_alias(%{__struct__: Macro.Env, aliases: aliases}, atom) when is_atom(atom),
     do: Keyword.fetch(aliases, :"Elixir.#{atom}")
 
+  @doc """
+  Fetches the name that the given module or atom has been aliased as.
+
+  Returns `{:ok, name}` if the alias exists, `:error` otherwise.
```

Comment:
What happens if the same module has been aliased more than once? We could return a random one, but perhaps would be better to return a list? But if we return a list, then there is no need to have a `fetch_` based API. We could use an empty list to mean "none". Thoughts?

Reply:
Good point. Perhaps `lookup_aliases/2`?

```elixir
iex> alias Foo.Bar, as: Baz
iex> Macro.Env.lookup_aliases(__ENV__, Foo.Bar)
[:Baz]
iex> alias Foo.Bar, as: Buzz
iex> Macro.Env.lookup_aliases(__ENV__, Foo.Bar)
[:Baz, :Buzz]
iex> Macro.Env.lookup_aliases(__ENV__, Unknown)
[]
```

Reply:
or `lookup_alias_as`? Since `lookup_aliases` might be confused with `fetch_alias`, which does the lookup the other way.

Reply:
Good call on `lookup`, since we already use it. I think `lookup_aliased_as` or `lookup_alias_as` are both great. Whatever reads better for someone who actually knows English :D

Reply:
While I think `lookup_aliased_as` reads slightly better, let's go with `lookup_alias_as` since the other usage of `lookup` is `lookup_import` and not `lookup_imported`.

Reply:
Good call. :100: